sdk/src/vote_program.rs

```diff
@@ -272,6 +272,11 @@ pub fn process_vote(keyed_accounts: &mut [KeyedAccount], vote: Vote) -> Result<(
         Err(ProgramError::InvalidArgument)?;
     }
 
+    if keyed_accounts[0].signer_key().is_none() {
```

Comment:
is it error prone to ask that every program do this verification?   can we include rules like this in a MIF?

Reply:
Yes

Reply:
This particular instruction would still be an exception though. If the account owner authorizes a vote signer, then accounts[0] doesn't need to be signed. That signature will be in accounts[1].